gitlab/resource_gitlab_project_test.go

```diff
@@ -383,6 +383,35 @@ func TestAccGitlabProject_initializeWithoutReadme(t *testing.T) {
 	})
 }
 
+func TestAccGitlabProject_archiveOnDestroy(t *testing.T) {
+	var project gitlab.Project
+	rInt := acctest.RandInt()
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGitlabProjectDestroy,
```

Comment:
Doesn't this default `CheckDestroy` fail if the project is archived?

Reply:
Oh I see, since it was already destroyed in the last step, there is no longer a `gitlab_project` in the state, so this `CheckDestroy` is a no-op.

It would be less messy to write a new `CheckDestroy` instead of using a test step.

Reply:
Yes, good idea :) Just pushed your suggested approach 🎉